### PR TITLE
[doc] Updates docs for cleanly upgrading to release/v4.4 ESP-IDF SDK.

### LINKED ISSUE #13602: [esp32] Issue with `menuconfig`
#### Problem
Trying to build the all-clusters-app for ESP fails with `Failed to resolve component 'openthread'` on Mac:

```
$ idf.py menuconfig

Executing action: menuconfig
Running cmake in directory /Users/mturon/src/nest/connectedhomeip/turon/examples/all-clusters-app/esp32/build
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -DCCACHE_ENABLE=0 /Users/mturon/src/nest/connectedhomeip/turon/examples/all-clusters-app/esp32"...
-- Found Git: /usr/local/git/current/bin/git (found version "2.34.1.703.g22d0c6ccf7-goog") 
-- IDF_TARGET not set, using default target: esp32
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /Users/mturon/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Users/mturon/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Users/mturon/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Project version: test_event_4_2021_07_06-2338-ga11413e3d
-- Building ESP-IDF components for target esp32
CMake Error at /Users/mturon/tools/esp-idf/tools/cmake/build.cmake:186 (message):
  Failed to resolve component 'openthread'.
Call Stack (most recent call first):
  /Users/mturon/tools/esp-idf/tools/cmake/build.cmake:217 (__build_resolve_and_add_req)
  /Users/mturon/tools/esp-idf/tools/cmake/build.cmake:426 (__build_expand_requirements)
  /Users/mturon/tools/esp-idf/tools/cmake/project.cmake:396 (idf_build_process)
  CMakeLists.txt:34 (project)


-- Configuring incomplete, errors occurred!
See also "/Users/mturon/src/nest/connectedhomeip/turon/examples/all-clusters-app/esp32/build/CMakeFiles/CMakeOutput.log".
cmake failed with exit code 1
```

### PR DESCRIPTION
**Pull request by turon**
#### Problem
Fix #13602

#### Change overview
Update docs with how to cleanly upgrade an existing ESP-IDF to the latest version.
Update tag to use release/v4.4.

#### Testing
Doc only change.  CI.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@turon Please restyle :)

**Review by andy31415 — APPROVED** [review]

File: examples/all-clusters-app/esp32/README.md [hunk 1]
```diff
@@ -48,10 +48,20 @@ step. To install these components manually, follow these steps:
...
 
+    To update an existing esp-idf toolchain to v4.4:
+
+          $ cd ~/tools/esp-idf
+          $ git fetch origin
+          $ git checkout release/v4.4
+          $ git reset --hard origin/release/v4.4
+          $ git submodule update --init
+          $ git clean -fdx
+          $ ./install.sh
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:63]: I think there should be a mention of `install.sh` and `export.sh` in this file. Can you please remove the other occurrence if it isn't clear

**Review by dhrishi — DISMISSED** [review]
Adding 'Request changes' so that it doesn't get merged accidently

**Pull request merged**